salt/cloud/clouds/digital_ocean.py

```diff
@@ -414,12 +447,23 @@ def __query_node_data(vm_name):
             pass
         finally:
             raise SaltCloudSystemExit(str(exc))
-
-    for network in data['networks']['v4']:
-        if network['type'] == 'public':
-            ip_address = network['ip_address']
+     
+    # add DNS records, set ssh_host, default to first found IP, preferring IPv4 for ssh bootstrap script target
+    addr_families, dns_arec_types = (('v4', 'v6'), ('A', 'AAAA'))
+    arec_map = dict(zip(addr_families, dns_arec_types))
+    for facing, addr_family, ip_address in [(net['type'], family, net['ip_address'])
+                                            for family in addr_families
+                                            for net in data['networks'][family]]:
+        log.info('found {0} IP{1} interface for "{2}"'.format(facing, addr_family, ip_address))
+        dns_rec_type = arec_map[addr_family]
+        if facing == 'public':
+            dnsrv = None
+            if create_dns_record:
+                dnsrv = __add_dns_addr__(dns_rec_type, ip_address)
+            if 'ssh_host' not in vm_ or not vm_['ssh_host']:
+                vm_['ssh_host'] = ip_address
+    
     vm_['key_filename'] = key_filename
-    vm_['ssh_host'] = ip_address
```

Comment:
`vm_['ssh_host']` must be set before calling the `salt.cloud.bootstrap` function (next line after this one) as we need this value [here](https://github.com/saltstack/salt/blob/develop/salt/utils/cloud.py#L392). Can you rewrite this part so that this line won't overwrite your `for` loop changes in lines 454-465 but that we can also maintain current functionality?